lucene/core/src/test/org/apache/lucene/index/TestCustomTermFreq.java

```diff
@@ -458,4 +458,50 @@ public void testFieldInvertState() throws Exception {
 
     IOUtils.close(w, dir);
   }
+
+  // LUCENE-8947: Indexing fails with "too many tokens for field" when using custom term frequencies
```

Comment:
Maybe add `when using large enough custom term frequencies to overflow int on accumulation`?

Reply:
Yeah that is a more precise description. I just added it to the comment.